testbed/core/views.py

```diff
@@ -263,20 +338,19 @@ def following_collection(request, pk):
     Note: While the collection URL only appears in LOLA-authenticated Actor objects,
     the collection itself follows standard ActivityPub public access patterns.
     """
-    actor = get_object_or_404(Actor, pk=pk)
+    try:
+        actor = Actor.objects.get(pk=pk)
+    except Actor.DoesNotExist:
```

Comment:
It feels like a step back to have to go back to try/except for each failed 'get' and to have to do it each time an Actor is needed in a view.  This could be caught at another layer?  i can imagine a couple ways to solve this and there are probably more
* a postprocessor in the middleware list
* a decorator called "handle_errors" on the view methods, and the decorator calls the view method then checks for Actor.DoesNotExist or other errors

Reply:
Another style choice that's also a decorator... I've written decorators for views error cases before which make the code very readable, you've probably seen them - @staff_required or @auth_required - if following that pattern you'd have a decorator for @valid_actor_required and it would not only make sure the Actor ID is in supplied,  fetch the Actor but also pass to the view, then catch Actor.DoesNotExist error and handle appropriately.